specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+  [propget] HRESULT Host([out, retval] LPWSTR * host);
+
+  /// Port of the server that requested client certificate authentication.
+  [propget] HRESULT Port([out, retval] int* port);
+
+  /// Returns true if the server that issued this request is an http proxy.
+  /// Returns false if the server is the origin server.
+  [propget] HRESULT IsProxy([out, retval] BOOL* isProxy);
+
+  /// Returns the `ICoreWebView2CertificateAuthorityList`.
+  /// The list contains distinguished name of certificate authorities
+  /// allowed by the server.
+  [propget] HRESULT CertificateAuthorities([out, retval]
+      ICoreWebView2CertificateAuthorityList** certificateAuthorityList);
+
+  /// Returns the `ICoreWebView2ClientCertificateList` when client
+  /// certificate authentication is requested. The list contains mutually
+  /// trusted CA certificates.
+  [propget] HRESULT MutuallyTrustedCertificates([out, retval]
+      ICoreWebView2ClientCertificateList** clientCertificateList);
+
+  /// Returns the selected certificate.
+  [propget] HRESULT SelectedCertificate([out, retval]
+      ICoreWebView2ClientCertificate** certificate);
+
+  /// Sets the certificate to respond to the server.
+  [propput] HRESULT SelectedCertificate(
+      [in] ICoreWebView2ClientCertificate* certificate);
+
+  /// The state of the Client Certificate Request Response State.
+  /// See `COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE` for descriptions of states.
+  /// The default is COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_DEFAULT_DIALOG.
+  [propget] HRESULT ClientCertificateRequestResponseState(
+      [out, retval] COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE* state);
+
+  /// Sets the `Client Certificate Request Response State` property.
+  [propput] HRESULT ClientCertificateRequestResponseState(
+      [in] COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE state);
+
+  /// Returns an `ICoreWebView2Deferral` object.  Use this operation to
+  /// complete the event at a later time.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+```
+
+## .NET/ WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2ClientCertificateRequestedEventArgs;
+    runtimeclass CoreWebView2ClientCertificate;
+
+    enum CoreWebView2ClientCertificateRequestResponseState
+    {
+        AbortRequest = 0,
+        ContinueWithoutCertificate = 1,
+        ContinueWithCertificate = 2,
+        ContinueWithDefaultDialog = 3,
+    };
+    enum CoreWebView2ClientCertificateKind
+    {
+        SmartCard = 0,
+        Pin = 1,
+        Other = 2,
+    };
+
+    runtimeclass CoreWebView2ClientCertificateRequestedEventArgs
+    {
+        // ICoreWebView2ClientCertificateRequestedEventArgs members
+        String Host { get; };
+        Int32 Port { get; };
+        Boolean IsProxy { get; };
+        IVector<string> CertificateAuthorities { get; };
```

Comment:
IVector or IVectorView? (What did we decide in earlier reviews?)

IVectorView aligns with the C++ interface. IVector is more generous, but may also create the impression that editing the list has some effect on the connection.

Reply:
We used IVectorView in the frame info. Let's use IVectorView here

Reply:
Used IVectorView for `AllowedCertificateAuthorities` and `MutuallyTrustedCertificates`.